dlfcn.c

```diff
@@ -239,15 +239,15 @@ void *dlopen( const char *file, int mode )
     {
         HANDLE hCurrentProc;
         DWORD dwProcModsBefore, dwProcModsAfter;
-        char lpFileName[MAX_PATH];
+        char *lpFileName;
         size_t i, len;
 
         len = strlen( file );
+        lpFileName = malloc( len + 1 );
 
-        if( len >= sizeof( lpFileName ) )
+        if( ! lpFileName )
         {
-            SetLastError( ERROR_FILENAME_EXCED_RANGE );
-            save_err_str( file );
+            SetLastError( ERROR_NOT_ENOUGH_MEMORY );

```

Comment:
You have removed `save_err_str` call, therefore error is not propagated. And `SetLastError` without `save_err_str` does nothing. So this branch when `lpFileName` is NULL is broken.